lib/session.js

```diff
@@ -24,6 +25,7 @@ module.exports = class Session {
     this[cookieSignerKey] = cookieSigner
     this[requestKey] = request
     this[sessionIdKey] = sessionId
+    this[encryptedSessionIdKey] = (prevSession && prevSession[sessionIdKey] === sessionId && prevSession[encryptedSessionIdKey]) || cookieSigner.sign(this.sessionId)
```

Comment:
@rclmenezes 
I deviated from your PR. If the prevSession has sessionIdKey and sessionId is the same, then we dont need to encrypt again, as it should be the same. Do you agree?

Reply:
Depends. If there are multiple secrets, I think anything with the old secret should be resigned with the new secret. Hence this PR:
https://github.com/fastify/session/pull/118

But honestly, I don't feel very strongly about it 🤷‍♂️

Reply:
For multiple unit tests we have this unit test

https://github.com/fastify/session/blob/b7661cdc46c228547d36fe6cc50a8e1190131b13/test/base.test.js#L79

So we have that case covered.

Reply:
Unfortunately, this test doesn't properly cover that case.

Adapt the test from my original PR and we'll see a failure:

```typescript
test('should support multiple secrets', async (t) => {
  t.plan(4)
  const sign = require('@fastify/cookie/signer').sign
  const newSecret = 'geheim'
  const options = {
    secret: [newSecret, DEFAULT_SECRET]
  }

  const sessionId = 'aYb4uTIhdBXCfk_ylik4QN6-u26K0u0e'
  const sessionIdSignedWithOldSecret = sign(sessionId, DEFAULT_SECRET)
  const sessionIdSignedWithNewSecret = sign(sessionId, newSecret)
  const htmlEncodedSessionIdSignedWithNewSecret = 'aYb4uTIhdBXCfk_ylik4QN6-u26K0u0e.InCp31AuDa7DX%2F8rGBz8RMFiCpmUtjcF%2BS7Aco7tur8'

  const plugin = fastifyPlugin(async (fastify, opts) => {
    fastify.addHook('onRequest', (request, reply, done) => {
      request.sessionStore.set(sessionId, {
        sessionId,
        encryptedSessionId: sessionIdSignedWithOldSecret
      }, done)
    })
  })
  function handler (request, reply) {
    request.session.test = {}
    reply.send(200)
  }
  const fastify = await buildFastify(handler, options, plugin)
  t.teardown(() => fastify.close())

  const responseForOldSecret = await fastify.inject({
    url: '/',
    headers: {
      'x-forwarded-proto': 'https',
      cookie: `sessionId=${sessionIdSignedWithOldSecret}; Path=/; HttpOnly; Secure`
    }
  })

  t.equal(responseForOldSecret.statusCode, 200)
  // It will be replaced with the new secret!
  t.equal(responseForOldSecret.headers['set-cookie'].includes(htmlEncodedSessionIdSignedWithNewSecret), true)

  const responseForNewSecret = await fastify.inject({
    url: '/',
    headers: {
      'x-forwarded-proto': 'https',
      cookie: `sessionId=${sessionIdSignedWithNewSecret}; Path=/; HttpOnly; Secure`
    }
  })
  t.equal(responseForNewSecret.statusCode, 200)
  t.equal(responseForNewSecret.headers['set-cookie'].includes(htmlEncodedSessionIdSignedWithNewSecret), true)
})
```

Why? What's the difference between this and the current test?

The current test does not set `encryptedSessionId` in the session prior to running the test. So `encryptedSessionId` never gets copied over here:
https://github.com/fastify/session/blob/master/lib/session.js#L31-L33

Since `encryptedSessionId` isn't set, it gets a new signature.

In production, `encryptedSessionId` would be in the previous session.

Reply:
I was running on main, not on this branch.

On this branch, we would need to remove `sessionId` and `encryptedSessionId` from the store. Otherwise we'll see the setter error.

But -- it does pass on your branch already so we're all set :)

Reply:
so you would approve this PR? Or am I missing something?

Reply:
Sorry sorry it's been a pretty terrible day. Reviewing now.